src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -327,11 +333,23 @@ public String login(@ModelAttribute("authorizeCommand") @Validated AuthorizeRequ
                     );
                     params.putIfAbsent(RESPONSE_TYPE, "code");
                     params.putIfAbsent(SCOPE, "openid profile roles");
+
                     responseUrl = spiService.authorize(params, cookie);
                 }
-                if (responseUrl != null && !responseUrl.contains("error")) {
+                final boolean loginSucces = responseUrl != null && !responseUrl.contains("error");
+                final boolean policyCheckFailed;
+                if (loginSucces) {
+                    policyCheckFailed = !policyService.evaluatePoliciesForUser(responseUrl, cookie, ipAddress);
+                } else {
+                    policyCheckFailed = false;
+                }
+                if (loginSucces && !policyCheckFailed) {
                     response.addHeader(HttpHeaders.SET_COOKIE, makeCookieSecure(cookie));
                     nextPage = "redirect:" + responseUrl;
+                } else if (policyCheckFailed) {
```

Comment:
Think this info message might be a bit misleading when loginSuccess == false (because that will be the reason policyCheckFailed is false). Might be better as
`} else if (loginSuccess && policyCheckFailed) {` ?

Reply:
I think this method is just getting too large in general, if we can split the checks into sub methods like, doLogin, evaluatePolicies it would greatly improve readability